commercialNetwork/src/js/initSlider.js

```diff
@@ -0,0 +1,201 @@
+      indexActiveSlide--
+      track.style.transform = shift(indexActiveSlide, zdvig)
+      document.body.classList.add("animation")
+
+      if (text) redraw("next", text, indexActiveSlide)
+    }
+
+    if (Math.abs(indexActiveSlide) === items.length - countActiveSlide) {
+      btnNext.classList.add("hiddenBtn-right")
+    }
+
+    currentPosition = zdvig * -indexActiveSlide
+
+    if (indexActiveSlide !== 0) btnPrev.classList.remove("hiddenBtn-left")
+
+    setTimeout(() => {
+      document.body.classList.remove("animation")
+    }, 300)
+  })
+
+  btnPrev.addEventListener("click", () => {
+    const zdvig = shiftCalculate()
+    const countActiveSlide = activeSlides(slider)
+
+    if (indexActiveSlide !== 0) {
+      indexActiveSlide++
+      track.style.transform = shift(indexActiveSlide, zdvig)
+      document.body.classList.add("animation")
+
+      if (text) redraw("prev", text, indexActiveSlide)
+
+      if (indexActiveSlide === 0) btnPrev.classList.add("hiddenBtn-left")
+
+      if (indexActiveSlide !== items.length - countActiveSlide) {
+        btnNext.classList.remove("hiddenBtn-right")
+      }
+    }
+
+    currentPosition = zdvig * -indexActiveSlide
+
+    setTimeout(() => {
+      document.body.classList.remove("animation")
+    }, 300)
+  })
+
+  function onDragStart(event) {
+    event.preventDefault()
+    isDrag = true
+
+    track.style.transitionDelay = "0s"
+
+    positionStart = event.type === "touchstart" ? event.touches[0].clientX : event.clientX
+    positionStart += currentPosition
+  }
+
+  function activeSlides(selector) {
+    const sliderWidth = selector.offsetWidth
+    const itemWidth = items[0].offsetWidth
+    return Math.floor(sliderWidth / itemWidth)
+  }
+
+  function onDragOver(event) {
+    if (!isDrag) return
+
+    const move = event.type === "touchmove" ? event.touches[0].clientX : event.clientX
+
+    track.style.transform = `translate3d(${move - positionStart}px, 0px, 0px)`
+
+    track.style.transitionDuration = "0ms"
+  }
+
+  function onDragEnd(event) {
+    if (!isDrag) return
+
+    const zdvig = shiftCalculate()
+    const countActiveSlide = activeSlides(slider)
+
+    let positionEnd = event.type === "touchend" ? event.changedTouches[0].clientX : event.clientX
+
+    positionEnd += currentPosition
+
+    if (
+      positionStart > positionEnd &&
+      Math.abs(positionEnd - positionStart) > zdvig * 0.2 &&
+      -indexActiveSlide !== items.length - countActiveSlide
+    ) {
+      if (
+        positionStart - positionEnd - 2 * zdvig > 0 &&
+        Math.abs(indexActiveSlide) + 2 < items.length - countActiveSlide
+      ) {
+        indexActiveSlide -= 2
+        if (text) redraw("next", text, indexActiveSlide)
+      } else {
+        indexActiveSlide--
+        if (text) redraw("next", text, indexActiveSlide)
+      }
+
+      document.body.classList.add("animation")
+    }
+
+    if (positionStart < positionEnd && positionEnd - positionStart > zdvig * 0.2 && indexActiveSlide !== 0) {
```

Comment:
const ... = positionEnd - positionStart > zdvig * 0.2
const ... = indexActiveSlide !==
 0.2 is constant